base/inference.jl

```diff
@@ -1411,7 +1469,15 @@ function abstract_call(f::ANY, fargs, argtypes::Vector{Any}, vtypes::VarTable, s
     tm = _topmod(sv)
     if length(argtypes)>2 && argtypes[3] ⊑ Int
         at2 = widenconst(argtypes[2])
-        if (at2 <: Tuple ||
+        if at2 <: SimpleVector && istopfunction(tm, f, :getindex)
+            if isa(argtypes[2], Const) && isa(argtypes[3], Const)
+                svecval = argtypes[2].val
+                idx = argtypes[3].val
+                if isa(idx, Int) && 1 <= idx <= length(svecval)
```

Comment:
Ah, true. Sight, the really annoying part here is going to be that for every extra case here, you need to duplicate it and add a special case for inlineable too since they need to model each other (for example, https://github.com/JuliaLang/julia/pull/20006/files#diff-c0d983631c937ee5acfd0137cd7332deR1515)